homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -195,22 +196,30 @@ def _async_register_button_devices(
         if "serial" not in device or device["serial"] in seen:
             continue
         seen.add(device["serial"])
-
-        dr_device = device_registry.async_get_or_create(
-            name=device["name"],
-            suggested_area=device["name"].split("_")[0],
-            manufacturer=MANUFACTURER,
-            config_entry_id=config_entry_id,
-            identifiers={(DOMAIN, device["serial"])},
-            model=f"{device['model']} ({device['type']})",
-            via_device=(DOMAIN, bridge_device["serial"]),
-        )
-
+        device_args = {
+            "name": device["name"],
+            "manufacturer": MANUFACTURER,
+            "config_entry_id": config_entry_id,
+            "identifiers": {(DOMAIN, device["serial"])},
+            "model": f"{device['model']} ({device['type']})",
+            "via_device": (DOMAIN, bridge_device["serial"]),
+        }
+        if "_" in device["name"]:
```

Comment:
We have this check on line 218 too.

Reply:
I figure we don’t want to suggest Unassigned as an area since it will create a new area they will have to delete later